### ISSUE TITLE
Warning dialog of Add-on Store is shown again although "Don't show again" checkbox has been checked

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#15261]
### Steps to reproduce:

1. Configure NVDA not to save settings on exit, and save config
2. Create a test profile and activate it manually
3. Launch the Add-on Store
4. When the Warning dialog is shown, check "Don't show again" checkbox and validate
5. Exit the Store
5.1 Disable the test profile (thus turn back to default profile)
6. Launch the Store
7.  When the Warning dialog is shown, check "Don't show again" checkbox and validate
8. Exit the store and restart NVDA
9. Work on something else and modify various NVDA settings, e.g. doc formatting settings. These modifications are not meant to be saved since NVDA config is not automatically saved.
10. Launch the Store
11. When the Warning dialog is shown, check "Don't show again" checkbox and validate
12. Save your config to avoid being bothered again with the Warning dialog

### Actual behavior:
Step 6: the warning dialog is shown again.
Step 10: the warning dialog is shown again.
Step 12: To be able to save the "Don't show again" status, the user also needs to save doc formatting settings are saved
### Expected behavior:
Step 6: The warning should not be shown because the "Don't show again" checkbox is not expected to be profile dependent.
Step 10: The warning should not be shown because the "Don't show again" checkbox should have been taken into account whether the user chooses to save the config or not .
Step 12: The user has had to save doc formatting settings whereas they didn't want to.

To summarize, memorizing the "Don't show again" choice:
* should not be profile dependent
* should be always saved independently of other NVDA settings

### Notes
@michaelDCurran has objected in https://github.com/nvaccess/nvda/pull/15246#issuecomment-1664737995 that other dialog's status is saved the same way in the config. However, the case of the Add-on Store's warning causes more issues because:
* Welcome and stat dialogs are usually shown when NVDA is first started; thus, there are not yet profile enabled and the auto save is enabled by default
* Not checked, but I think that the setting for these dialogs are saved in a config section that exists only in default profile
* In NVDA, there are probably other issues of settings that can be saved in profile whereas the use case is probably not applicable to profiles

### NVDA logs, crash dumps and other attachments:
N/A
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2023.2beta1
#### Windows version:
Windows 10 2004 (AMD64) build 19041.388
#### Name and version of other software in use when reproducing the issue:
N/A
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
N/A
#### If NVDA add-ons are disabled, is your problem still occurring?
N/A
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A

**Comment by XLTechie** [comment]
@CyrilleB79 Can you please go through (follow) your STR and verify that this is indeed what happens? Because following exactly your STR through step 7, it isn't for me what happens.

When I open the store again in step 6, no warning dialog is shown. So your step 7 does not apply.

This is expected, because the testing profile is still active, and the warning has been turned off in the profile.
If I return to normal profile before step 6, then step 7 happens as you describe, which proves that this setting is profile aware, which shouldn't be the case. But in your STR you do not deactivate the profile, so step 7 shouldn't happen that way for you.

Also, your step 12 has the expected result. If you modified doc settings in step 9, and then save your config in step 12, I would expect doc settings to be saved, and that is what you say happens.
So I don't understand your complaint about step 12.

**Comment by XLTechie** [comment]
To me, the real issue here can be summed up by following your steps 1, 2, 3, 4, 8, and 10.

That is enough to prove that the "do not ask again" is profile aware, which shouldn't be the case.
Anything else is a side effect of that incorrect initial condition.

**Comment by Brian1Gaff** [comment]
I have to say that this dialog and other behaviours does seem somewhat counter intuitive. Its as if the new don't show again is actually not looking at the right things, or is in the wrong place, but then again, if its to warn about add ons, it has to be there for a new user. Should really this message not be always there, maybe not so verbose, as it really needs everyone to realise that there could be security issues. What worries me a little is that whereas we got system admins to trust nvda, in effect we have now removed that safeguard and they may revert to feeling the program is a risk.
 Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.

**Comment by Brian1Gaff** [comment]
So instead of checking a flag in the main config as well as the one you are in it really should do the same in the non active ones.
 However the warning in add ons I still feel should appear every time.
 Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.

**Comment by XLTechie** [comment]
@Brian1Gaff system admins are not, in general, that stupid.

For one thing, most of them don't allow add-ons in the first place, at least as 
I understand the usage patterns of restricted corporate installs.

Secondly, if they are trusting add-ons, a generic warning is highly likely to be 
recognized for what it is--especially since the same text appears in the user 
guide, which they have presumably read recently.

Windows is full of "don't do this unless you want to risk that" type messages 
that appear as "don't show this again" dialogs.
The paradigm is well established.

As for having the warning dialog appear all the time: that would be 
disruptive for normal users, and you may be sure that an add-on would be 
released in no time flat to suppress it if it was shown all the time.

And no, it shouldn't check a flag in the non-active profiles as well. It should 
ignore profiles entirely, and effect, and be read from, the normal (global) 
config only.

**Comment by CyrilleB79** [comment]
> @CyrilleB79 Can you please go through (follow) your STR and verify that this is indeed what happens? Because following exactly your STR through step 7, it isn't for me what happens.

@XLTechie, good catch. A step was missing to disable the test profile, I have added 5.1



> To me, the real issue here can be summed up by following your steps 1, 2, 3, 4, 8, and 10.
> That is enough to prove that the "do not ask again" is profile aware, which shouldn't be the case.

The current issue not only points the problem with profiles, but also the problem occurring due to the fact that the config should be saved to disk so that the choice of the user is memorized next time NVDA restarts. Hence the comments for steps 10 and 12.

**Comment by seanbudd** [comment]
I think the ideal solution here is to maintain a separate config file for settings which should always write to disk immediately and not participate in profiles.

**Comment by CyrilleB79** [comment]
Since #16636 was merged, the status of this warning checkbox is not profile dependent anymore. The issue with the warning turning back when the config is not saved still stands.
